src/d/actor/d_a_shutter.cpp

```diff
@@ -5,80 +5,292 @@
 
 #include "d/actor/d_a_shutter.h"
 #include "d/d_procname.h"
+#include "d/d_com_inf_game.h"
+
+const float daShutter_c::m_max_speed[2] = {3.0f, 3.0f};
+const float daShutter_c::m_min_speed[2] = {1.0f, 1.0f};
+const float daShutter_c::m_move_len[2] = {200.0f, 84.0f};
+const float daShutter_c::m_width[2] = {220.0f, 82.0f};
+const float daShutter_c::m_height[2] = {200.0f, 100.0f};
+const s16 daShutter_c::m_bdlidx[2] = {4, 4};
+const s16 daShutter_c::m_dzbidx[2] = {7, 7};
+const s32 daShutter_c::m_heapsize[2] = {0x1140, 0x3000};
+const Vec daShutter_c::m_cull_min[2] = {{-500.0f, -100.0f, -50.0f}, {-150.0f, -100.0f, -50.0f}};
+const Vec daShutter_c::m_cull_max[2] = {{500.0f, 250.0f, 50.0f}, {150.0f, 250.0f, 50.0f}};
+
+char* daShutter_c::m_arcname[2] = {"Htobi1", "Htobi2"};
+char* daShutter_c::m_open_ev_name[2] = {"HYSDOOROPEN", "R03DOOROPEN"};
+char* daShutter_c::m_close_ev_name[2] = {0, "R03DOORCLOSE"};
+char* daShutter_c::m_staff_name[2] = {"Htobi1", "Htobi2"};
 
 /* 00000078-00000108       .text _delete__11daShutter_cFv */
 bool daShutter_c::_delete() {
-    /* Nonmatching */
+    dComIfG_resDelete(&mPhs, m_arcname[mType]);
+    if (heap != NULL) {
+        for (int i = 0; i < (int)ARRAY_SIZE(mMtx); i++) {
+            dComIfG_Bgsp()->Release(mdBgW[i]);
+        };
+    }
+    return TRUE;
 }
 
 /* 00000108-00000128       .text CheckCreateHeap__FP10fopAc_ac_c */
-static BOOL CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CheckCreateHeap(fopAc_ac_c* i_this) {
+    return ((daShutter_c*)i_this)->CreateHeap();
 }
 
 /* 00000128-000002B8       .text CreateHeap__11daShutter_cFv */
-void daShutter_c::CreateHeap() {
-    /* Nonmatching */
+BOOL daShutter_c::CreateHeap() {
+    J3DModelData* modelData = (J3DModelData *)dComIfG_getObjectRes(m_arcname[mType], m_bdlidx[mType]);
+    JUT_ASSERT(0x121, modelData != 0);
+    for (int i = 0; i < (int)ARRAY_SIZE(mMtx); i++) {
+        mpModel[i] = mDoExt_J3DModel__create(modelData, 0x80000, 0x11000022);
+        if (!mpModel[i]) {
+            return FALSE;
+        }
+        mdBgW[i] = new dBgW();
+        if (mdBgW[i]) {
+            cBgD_t* pData = (cBgD_t *)dComIfG_getObjectRes(m_arcname[mType], m_dzbidx[mType]);
+            if (mdBgW[i]->Set(pData, 0x1, &mMtx[i]) == 1) {
+                return FALSE;
+            }
+        }
+        else {
+            return FALSE;
+        }
+    }
+    return TRUE;
 }
 
 /* 000002B8-000004B4       .text Create__11daShutter_cFv */
 s32 daShutter_c::Create() {
-    /* Nonmatching */
+    fopAcM_SetMtx(this, mpModel[0]->getBaseTRMtx());
+    Vec cullMin = m_cull_min[mType];
+    Vec cullMax = m_cull_max[mType];
+    fopAcM_setCullSizeBox(this, cullMin.x, cullMin.y, cullMin.z, cullMax.x, cullMax.y, cullMax.z);
+    mNo = daShutter_prm::getmNo(this);
```

Comment:
```suggestion
    mSwitchNo = daShutter_prm::getSwitchNo(this);
```
Member name of params should generally also match the inline name if it makes sense